packages/opentelemetry-core/src/internal/validators.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const KEY_MAX_SIZE = 256;
+const VALUE_MAX_SIZE = 256;
+
+/**
+ * Key is opaque string up to 256 characters printable. It MUST begin with a
+ * lowercase letter, and can only contain lowercase letters a-z, digits 0-9,
+ * underscores _, dashes -, asterisks *, and forward slashes /.
+ */
+export function validateKey(key: string): boolean {
```

Comment:
Would it be possible to write this as a regex?

Same question about `validateValue` below.

Reply:
It is possible but it's not pretty. Here's a golang example: https://github.com/lightstep/tracecontext.go/blob/master/tracestate/package.go#L26

Reply:
`/^[a-z][_0-9a-z-*/]{0,255}$/` this is how regex looks like to validate keys. As @bg451 mentioned, regex for values are little complex and not pretty.

Reply:
That regex to me looks like a regex for the whole of tracestate, which ends up being pretty complex, but since we are splitting into keys and values, we can be simpler.